src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1306,3 +1393,69 @@ int sd_dhcp_server_set_relay_target(sd_dhcp_server *server, const struct in_addr
         server->relay_target = *address;
         return 1;
 }
+
+int sd_dhcp_server_add_static_lease(
+                sd_dhcp_server *server,
+                const struct in_addr* address,
+                uint8_t *mac_addr,
+                size_t mac_addr_size) {
+         _cleanup_(dhcp_lease_freep) DHCPLease *lease = NULL, *old = NULL;
+        int r;
+
+        assert_return(server, -EINVAL);
+        assert_return(address, -EINVAL);
+        assert_return(mac_addr, -EINVAL);
+        assert_return(mac_addr_size == ETH_ALEN, -EINVAL);
+
+        lease = new(DHCPLease, 1);
+        if (!lease)
+                return -ENOMEM;
+
+        *lease = (DHCPLease) {
+                .address = address->s_addr,
+                .client_id.length = mac_addr_size,
+                .gateway = 0,
+                .expiration = 0,
+        };
+        lease->client_id.data = memdup(mac_addr, mac_addr_size);
+        if (!lease->client_id.data)
+                return -ENOMEM;
+
+        r = hashmap_ensure_allocated(&server->static_leases_by_client_id, &dhcp_lease_hash_ops);
+        if (r < 0)
+                return r;
+
+        old = hashmap_get(server->static_leases_by_client_id, &lease->client_id);
+        r = hashmap_replace(server->static_leases_by_client_id, &lease->client_id, lease);
+        if (r < 0) {
+                TAKE_PTR(old);
+                return r;
+        }
+
+        TAKE_PTR(lease);
+
+        return 0;
+}
+
+int sd_dhcp_server_remove_static_lease(sd_dhcp_server *server, uint8_t *mac_addr, size_t mac_addr_size) {
```

Comment:
It is not necessary to introduce this function. Please implement this feature in `sd_dhcp_server_add_static_lease()` when `address` is `NULL` or empty. And, maybe it is better to rename `sd_dhcp_server_add_static_lease()` -> `sd_dhcp_server_set_static_lease()`.

Reply:
I I understand this [comment](https://github.com/systemd/systemd/pull/15556#discussion_r562084590) correnctly, I was specifically requested not to do this?